docs/JSON-RPC.md

```diff
@@ -0,0 +1,435 @@
+
+- A **request** from the consumer to Jamulus.
+- A **response** from Jamulus to the consumer.
+- A **notification** from Jamulus to the consumer.
+
+## Example
+
+After opening a TCP connection to the JSON-RPC server, the connection must be authenticated:
+
+```json
+{"id":1,"jsonrpc":"2.0","method":"jamulus/apiAuth","params":{"secret": "...the secret from the file in --jsonrpcsecretfile..."}}
+```
+
+Request must be sent as a single line of JSON-encoded data, followed by a newline character. Jamulus will send back a **response** in the same manner:
+
+```json
+{"id":1,"jsonrpc":"2.0","result":"ok"}
+```
+After successful authentication, the following **request** can be sent:
+
+```json
+{"id":2,"jsonrpc":"2.0","method":"jamulus/getMode","params":{}}
+```
+
+The request must be sent as a single line of JSON-encoded data, followed by a newline character. Jamulus will send back a **response** in the same manner:
+
+```json
+{"id":2,"jsonrpc":"2.0","result":{"mode":"client"}}
+```
+
+Jamulus will also send **notifications** to the consumer:
+
+```json
+{"jsonrpc":"2.0","method":"jamulusclient/chatTextReceived","params":{"text":"<font color=\"mediumblue\">(01:23:45 AM) <b>user</b></font> test"}}
+```
+
+## Method reference
+### jamulus/apiAuth
+
+Authenticates the connection which is a requirement for calling further methods.
+
+Parameters:
+
+| Name | Type | Description |
+| --- | --- | --- |
+| params | object | No parameters (empty object). |
+
+Results:
+
+| Name | Type | Description |
+| --- | --- | --- |
+| result | string | "ok" on success |
+
+
+### jamulus/getMode
+
+Returns the current mode, i.e. whether Jamulus is running as a server or client.
+
+Parameters:
+
+| Name | Type | Description |
+| --- | --- | --- |
+| params | object | No parameters (empty object). |
+
+Results:
+
+| Name | Type | Description |
+| --- | --- | --- |
+| result.mode | string | The current mode (server or client). |
+
+
+### jamulus/getVersion
+
+Returns Jamulus version.
+
+Parameters:
+
+| Name | Type | Description |
+| --- | --- | --- |
+| params | object | No parameters (empty object). |
+
+Results:
+
+| Name | Type | Description |
+| --- | --- | --- |
+| result.version | string | The Jamulus version. |
+
+
+### jamulusclient/getChannelInfo
+
+Returns your profile information.
+
+Parameters:
+
+| Name | Type | Description |
+| --- | --- | --- |
+| params | object | No parameters (empty object). |
+
+Results:
+
+| Name | Type | Description |
+| --- | --- | --- |
+| result.name | string | Your name. |
+| result.skillLevel | string | Your skill level (beginner, intermediate, expert, or null). |
+
+
+### jamulusclient/getClientInfo
+
+Returns the client information.
+
+Parameters:
+
+| Name | Type | Description |
+| --- | --- | --- |
+| params | object | No parameters (empty object). |
```

Comment:
This should at least say what the object actually contains.

Reply:
Argh. Sorry I thought there was an output of a JSON object with data not specified here. I misread the "empty" so this comment is invalid